src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,603 @@
+          rsc_outline_dir = relpathalone(ctx.rsc_outline_dir)
+          args = [
+            # NB: Without this setting, rsc will be missing some symbols
+            #     from the scala library.
+            '--include-scala-library-synthetics', # TODO generate these once and cache them
+            # NB: We need to add these extra dependencies in order to be able
+            #     to find symbols used by the scalac jars.
+            '--dependency-classpath', os.pathsep.join(scalac_classpath_path_entries),
+            # NB: The directory to dump the semanticdb jars generated by metacp.
+            # TODO: break this out into a separate job so that we can apply
+            #       it once per 3rdparty target.
+            '--out', rsc_outline_dir,
+            os.pathsep.join(cp_entries),
+          ]
+          metacp_wu = self._runtool(
+            'scala.meta.cli.Metacp',
+            'metacp',
+            args,
+            distribution,
+            tgt=tgt,
+            input_files=(scalac_classpath_path_entries + classpath_rel),
+            output_dir=rsc_outline_dir)
+          metacp_stdout = stdout_contents(metacp_wu)
+          metacp_result = json.loads(metacp_stdout)
+          metai_classpath = []
+          
+          def desandboxify_pantsd_loc(path):
+            # TODO come up with a cleaner way to maybe relpath paths.
+            try:
+              path = relpathalone(path)
+            except Exception:
+              pass
+            pattern = 'process-execution[^{}]+/'.format(re.escape(os.path.sep))
+            return re.split(pattern, path)[-1]
+
+          # TODO when these are generated once, we won't need to collect them here.  
+          metai_classpath.append(desandboxify_pantsd_loc(metacp_result["scalaLibrarySynthetics"]))
+          # NB The json is absolute pathed pointing into either the buildroot or
+          #    the temp directory of the hermetic build. This relativizes the keys.
+          status_elements = {
+            desandboxify_pantsd_loc(k): v
+            for k,v in metacp_result["status"].items()
+          }
+
+          for cp_entry in classpath_rel:
+            metai_classpath.append(desandboxify_pantsd_loc(status_elements[cp_entry]))
+          for cp_entry in jvm_lib_jars_abs:
+            metai_classpath.append(desandboxify_pantsd_loc(status_elements[cp_entry]))
+
+          metacp_classpath_entries = metai_classpath
+
+          # Step 1.5: metai Index the semanticdbs
+          # -------------------------------------
+          # TODO have metai write to a different spot than metacp
+          # Currently, the metai step depends on the fact that materializing
+          # ignores existing files. It should write the files to a different
+          # location, either by providing inputs from a different location,
+          # or invoking a script that does the copying
+          args = [
+            '--verbose',
```

Comment:
I noticed that you're not using `--verbose` in Metacp invocations, but using it in Metai invocations. Is this deliberate?

Reply:
No. Would it be better to use it in all cases? I'm not sure what it does--besides the obvious.

Reply:
In both Metacp and Metai, `--verbose` does the same thing - every now and then, prints progress to stderr. I introduced it to keep better track of long-running operations for my personal experiments, but I don't know whether doing this is idiomatic in Pants.